### PR TITLE
[NFC] lifetime cleanup

### PR DESCRIPTION
**Pull request by lxfind**
Remove unnecessary lifetime.
Remove unnecessary mustuse.

### CODE REVIEW

File: crates/sui-storage/src/write_ahead_log.rs [hunk 1]
```diff
@@ -66,7 +66,6 @@ pub trait WriteAheadLog<'a, C> {
     ///   proceeding.
     ///
     ///   Err(e) => An error occurred.
-    #[must_use]
```
Comment (mystenmark) [inline crates/sui-storage/src/write_ahead_log.rs:69]: Why remove this? I can't think of a situation where you would call `begin_tx` and not hold the result.

Comment (mystenmark) [inline crates/sui-storage/src/write_ahead_log.rs:69]: Why remove this? I can't think of a case where you'd call begin_tx and not hold the result.

**Review by mystenmark — CHANGES_REQUESTED** [review]

File: crates/sui-storage/src/write_ahead_log.rs (hunk 1 above)
Comment (lxfind) [reply crates/sui-storage/src/write_ahead_log.rs:69]: `Result` is already a `#[must_use]` type. This annotation is unnecessary.

**Review by velvia — APPROVED** [review]
Thanks!

**Pull request merged**